polar_route/mesh_generation/environment_mesh.py

```diff
@@ -157,8 +161,136 @@ def to_geojson(self):
+                    GT[1] w-e pixel resolution / pixel width.
+                    GT[2] row rotation (typically zero).
+                    GT[3] y-coordinate of the upper-left corner of the upper-left pixel.
+                    GT[4] column rotation (typically zero).
+                    GT[5] n-s pixel resolution / pixel height (negative value for a north-up image).
+
+            """
+            resx = (extent[2] - extent[0]) / ncols
+            resy = (extent[3] - extent[1]) / nlines
+            return [extent[0], resx, 0, extent[3] , 0, -resy]
+        def validate_params(params):
+          """
+                validates that the parameters needed for the export are not missing
+          """
+          if (params == None):
+             raise ValueError('Parameters missing! Can not save mesh in tif format with None parameters')
+          if ( "data_name" not in params.keys() ):
+             raise ValueError('data_name should be specified in the params while saving in tif format')
+          if ( "sampling_resolution" not in params.keys() ):
+             raise ValueError('sampling_resolution should be specified in the params while saving in tif format')
+
+        validate_params(params)
+        data_name = params["data_name"]
+        DEFAULT_PROJ = 4326
+        projection = DEFAULT_PROJ
+        if "projection" in params.keys():
+            projection = int (params["projection"])
+    
+        # Get image dimensions
+        nlines = params["sampling_resolution"][0]
+        ncols =  params["sampling_resolution"][1]
+
+        #define image extent based on mesh bounds
+        extent = [self.bounds.get_long_min(), self.bounds.get_lat_min(), self.bounds.get_long_max() , self.bounds.get_lat_max()]
+
+        logging.info ("Generating the tif image ...")
+        samples = generate_samples()
+        # create raster band and populate with sampled data of image_size (sampling_resolution)
+        # get GDAL driver GeoTiff
+        driver = gdal.GetDriverByName('GTiff')
+        # reading the samples value
+        data= []
+        data = np.reshape(np.asarray ([get_sample_value(sample) for sample in samples] , dtype=np.float32) , (nlines, ncols))
+        # create a temp grid
+        grid_data = driver.Create('grid_data', ncols, nlines, 1, gdal.GDT_Float32)
+        # setup geo-transform
+        grid_data.SetGeoTransform(get_geo_transform(extent, nlines, ncols))
+        # Write data 
+        srs = osr.SpatialReference() 
+        srs.ImportFromEPSG(DEFAULT_PROJ)
+        grid_data.SetProjection( srs.ExportToWkt())
+        grid_data.GetRasterBand(1).WriteArray(data)
+         
+        #check if color table is provided
+        if "color_table" in params.keys():
+            grid_data.GetRasterBand(1).SetRasterColorTable( params["color_table"] )
+ 
+        # Save the file
+        file_path = f'{path}mesh_{data_name}_{str(nlines)}x{str(ncols)}.tif'
```

Comment:
Might be best to just make this the default filename and use the user defined path when given? I put `"weddell_sic.tif"` in as the output path and got a file called `"weddell_sic.tifmesh_SIC_400x400.tif"` from the save method.